4_advanced_usage/40_node_properties.txt

```diff
@@ -0,0 +1,65 @@
+=== Node properties
+
+==== Defining properties
+
+Node properties can be found in the "properties" tab of each node in Rudder.
+
+Node properties can be modified using Rudder API, see http://www.rudder-project.org/rudder-api-doc/#api-Nodes-updateNodeProperties
+
+Currently a property is interpreted as a string value. So properties can be seen as a list of key=value pairs. Example: datacenter=Paris.
+
+==== Using properties
+
+You can use a node property almost everywhere in Rudder:
+- in directive parameters
+- in the technique editor
+- in your own techniques an generic methods code
+
+To use a property, simply use the variable node.properties with the variable call syntax.
+
+Example with a string property named datacenter:
+
+----
+
+${node.properties[datacenter]}
+
+----
+
+
+==== Under the hood
+
+On the server, one or more properties files are written for each node in the /var/rudder/share/<uuid>/rules/cfengine-community/properties.d/ directory.
+This directory is then synchronized by the agent with all other promise files.
+
+In the agent, properties are made available in the node.<namespace> container that contains the values.
+Those values are read from /var/rudder/cfengine-community/inputs/properties/*.json
+All files are taken in order and override the previous ones, the last one wins.
+
+Each file must contain at least a 2 levels JSON content, the first level is the namespace level
+and the second level is the key level.
+The namespace name must be an ASCII name that doesn't start with '_', (it must match the following regex: [a-zA-Z0-9][a-zA-Z0-9_]*).
```

Comment:
", (" is an odd use of punctuation :)

How about "The namespace name must be an ASCII name that doesn't start with '_' and must match the following regex: [a-zA-Z0-9][a-zA-Z0-9_]*." ?